src/DomainStorageServer.js

```diff
@@ -58,7 +60,7 @@ const getParentDomain = hostname => {
 
 const isMessageTrusted = (origin) => {
   // todo also verify the source of the message
-  return getParentDomain((new URL(origin)).hostname) === getParentDomain(window.location.hostname)
+  return _trustedDomains.includes(origin) || getParentDomain((new URL(origin)).hostname) === getParentDomain(window.location.hostname)
```

Comment:
`origin` here is expected as something like `https://www.example.com`. If you only want the domain then you'll need to use `(new URL(origin)).hostname`. Better store it in a const and use in both places